tests/create-apollo-server-setup.js

```diff
@@ -40,37 +43,44 @@ function setupApolloServerTests(loadApolloServer, clearCachedModules, options, a
       const instrumentationPlugin = createPlugin(nrApi.shim, pluginConfig)
 
       // Do after instrumentation to ensure express isn't loaded too soon.
-      const apolloServerPkg = loadApolloServer()
-      const { ApolloServer, gql } = apolloServerPkg
+      apolloServerPkg = loadApolloServer()
+      const { ApolloServer, gql, startStandaloneServer } = apolloServerPkg
       const schema = getTypeDefs(gql)
-      const errorSchema = setupErrorSchema(apolloServerPkg, resolvers)
+      const errorSchema = setupErrorSchema(apolloServerPkg, resolvers, isApollo4(apolloServerPkg))
 
       server = new ApolloServer({
         typeDefs: [schema, errorSchema],
         resolvers,
-        plugins: [...startingPlugins, instrumentationPlugin]
+        plugins: [...startingPlugins, instrumentationPlugin],
+        allowBatchedHttpRequests: true
       })
 
-      return server.listen({ port: 0 }).then(({ url }) => {
-        serverUrl = url
+      const { url } = startStandaloneServer
+        ? await startStandaloneServer(server, { listen: { port: 0 } })
+        : await server.listen({ port: 0 })
 
-        t.context.helper = helper
-        t.context.serverUrl = serverUrl
-      })
+      serverUrl = url
+
+      t.context.helper = helper
+      t.context.serverUrl = serverUrl
     })
 
     t.afterEach(() => {
-      server.stop()
+      helper.agent.errors.traceAggregator.clear()
+    })
+
+    t.teardown(async () => {
+      await server.stop()
 
       helper.unload()
       server = null
       serverUrl = null
       helper = null
 
-      clearCachedModules(['express', 'apollo-server'])
+      clearCachedModules(['express'])
```

Comment:
Curious why this moved to `tests/integration/apollo-server-setup.js`?

Reply:
I can restore here now that I added try catch in the clearCachedModules. It was failing when the packages don't exist